controller/files.php

```diff
@@ -45,6 +45,30 @@
 	/** @var ILogger */
 	private $logger;
 
+        /**
+         * @NoAdminRequired
+         *
+         * Returns a exif of picture
+         *
+         * @param string $location a path picture
+         * @return array
+         */
+        public function exif($location){
+            $filename=\OC\Files\Filesystem::getLocalFile($location);
+            $exif = false;
+	    $iptc = false;
+	    if (is_callable('exif_read_data')) {
+		$exif=@exif_read_data($filename);
```

Comment:
yes. for external "local" return direct path, for external nonlocal return path to cached file.